src/ngraph/op/generate_mask.hpp

```diff
@@ -0,0 +1,81 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <memory>
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/op.hpp"
+#include "ngraph/state/rng_state.hpp"
+namespace ngraph
+{
+    namespace op
+    {
+        /// \brief GenerateMask
+        ///
+        class GenerateMask : public op::Op
+        {
+        public:
+            GenerateMask(const std::shared_ptr<Node>& training,
+                         const std::shared_ptr<Node>& activate,
+                         const Shape& shape,
+                         const element::Type& element_type,
+                         double prob,
```

Comment:
Frameworks are very inconsistent about whether this is the keep or the drop probability, so we should call it `keep_probability` or `dropout_probability` to be clear.